### ISSUE TITLE
Decode is slow when using (headers: true)

### ISSUE THREAD

**Issue by baroldgene** [beatrichartz/csv#25]
When asking the CSV parser to add headers it seems to take a while to start.  Once going it is generally fast.  See attached gif for demo.  The **only** difference in the code between the two panes is ```(headers: true)```
![slowcsv](https://cloud.githubusercontent.com/assets/1310679/12407148/6f650dd8-be22-11e5-8319-935a41dd3c42.gif)

**Comment by baroldgene** [comment]
If it helps this is all done as a stream.  Here's the code:

```elixir
    File.stream!(options[:file])
     |> CSV.decode #(headers: true)
     |> Stream.map(&IO.inspect(&1))
     |> Stream.run
```

**Comment by beatrichartz** [comment]
Interesting, thanks for raising this! Will have a look soon

**Comment by beatrichartz** [comment]
I fixed this in `1.2.3`, please try again with this version - if you still have issues, feel free to reopen. Thanks again for reporting this!

**Comment by baroldgene** [comment]
Works great!  Thanks!
